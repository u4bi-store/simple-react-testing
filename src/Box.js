import React, { Component } from 'react'

class Box extends Component {

    state = {
        tick : 0,
        text : ''
    }

    addTick = _ => this.setState( ( { tick }) => ({ tick : ++tick }))

    setText = ({ target : { value }}) => this.setState({ text : value })

    render() {
        const { tick, text } = this.state
        const { addTick, setText } = this

        return (
        <div className='Box'>

            <span>{ tick }</span>

            <button onClick={ addTick }>AddTick</button>

            <input type='text' value={ text } onChange={ setText } />

        </div>
        )
    }
}

export default Box