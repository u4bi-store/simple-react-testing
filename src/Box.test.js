import React from 'react'
import { shallow } from 'enzyme'
import Box from './Box'

describe('Box', () => {
    let component = null

    it('has a React Element', () => {
        component = shallow(<Box />)
    })

    it('matches snapshot', () => expect(component).toMatchSnapshot())

    describe('addTick()', () => {

        it('has a span', () => expect(component.find('span').exists()).toEqual(true))

        it('has a tick value', () => expect(component.find('span').text()).toEqual('0'))

        it('has a button', () => expect(component.find('button').exists()).toEqual(true))

        it('simulates button click', () => {

            component.find('button')
                    .simulate('click')


            expect(component.state().tick).toEqual(1)
            expect(component.find('span').text()).toEqual('1')

        })

    })

    describe('setText()', () => {

        it('has a input', () => expect(component.find('input').exists()).toEqual(true))

        it('has a input type', () => expect(component.find('input').props().type).toEqual('text'))
    
        it('simulates input change', () => {

            const mock = {
                target : {
                    value : 'hello'
                }
            }

            component.find('input')
                    .simulate('change', mock)


            expect(component.state().text).toEqual('hello')
            expect(component.find('input').props().value).toEqual('hello')

        })

    })    

})