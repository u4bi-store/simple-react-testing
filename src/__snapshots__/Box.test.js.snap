// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<div
  className="Box"
>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    AddTick
  </button>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
